pkg/management/postgres/metrics/collector.go

```diff
@@ -378,6 +387,65 @@ func (c QueryCollector) collect(conn *sql.DB, ch chan<- prometheus.Metric) error
 	return nil
 }
 
+// isCollectable checks if a query to collect metrics can be executed or not.
+// The method tests the query provided in the PredicateQuery property within the same transaction
+// used to collect metrics.
+// Accepts a query that returns a boolean on the first column of the first row.
+func (c QueryCollector) isCollectable(tx *sql.Tx) (bool, error) {
+	if c.userQuery.PredicateQuery == "" {
+		return true, nil
+	}
+
+	rows, err := tx.Query(c.userQuery.PredicateQuery)
+	if err != nil {
+		return false, err
+	}
+
```

Comment:
was wondering why you're using `Query` rather than `QueryRow`, and I think it's because you want to query for `Columns()` below... but.
Is it worth it?

Reply:
Yep, you're right! I think there is something related to what we want to achieve. As said above we can assume that the predicate is one single row and deliver the `QueryRow` or be open on what the query provides and just have a look at the first row in the first column.